python/plugins/processing/algs/qgis/ServiceAreaFromLayer.py

```diff
@@ -98,62 +97,64 @@ def initAlgorithm(self, config=None):
+        params.append(QgsProcessingParameterField(self.DIRECTION_FIELD,
+                                                  self.tr('Direction field'),
+                                                  None,
+                                                  self.INPUT,
+                                                  optional=True))
+        params.append(QgsProcessingParameterString(self.VALUE_FORWARD,
+                                                   self.tr('Value for forward direction'),
+                                                   optional=True))
+        params.append(QgsProcessingParameterString(self.VALUE_BACKWARD,
+                                                   self.tr('Value for backward direction'),
+                                                   optional=True))
+        params.append(QgsProcessingParameterString(self.VALUE_BOTH,
+                                                   self.tr('Value for both directions'),
+                                                   optional=True))
+        params.append(QgsProcessingParameterEnum(self.DEFAULT_DIRECTION,
+                                                 self.tr('Default direction'),
+                                                 list(self.DIRECTIONS.keys()),
+                                                 defaultValue=2))
+        params.append(QgsProcessingParameterField(self.SPEED_FIELD,
+                                                  self.tr('Speed field'),
+                                                  None,
+                                                  self.INPUT,
+                                                  optional=True))
+        params.append(QgsProcessingParameterNumber(self.DEFAULT_SPEED,
+                                                   self.tr('Default speed (km/h)'),
+                                                   QgsProcessingParameterNumber.Double,
+                                                   5.0, False, 0, 99999999.99))
+        params.append(QgsProcessingParameterNumber(self.TOLERANCE,
+                                                   self.tr('Topology tolerance'),
+                                                   QgsProcessingParameterNumber.Double,
+                                                   0.0, False, 0, 99999999.99))
 
         for p in params:
             p.setFlags(p.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
             self.addParameter(p)
 
-        self.addOutput(OutputVector(self.OUTPUT_POINTS,
-                                    self.tr('Service area (boundary nodes)'),
-                                    datatype=[dataobjects.TYPE_VECTOR_POINT]))
-        self.addOutput(OutputVector(self.OUTPUT_POLYGON,
-                                    self.tr('Service area (convex hull)'),
-                                    datatype=[dataobjects.TYPE_VECTOR_POLYGON]))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT_POINTS,
```

Comment:
What do you think about making this and OUTPUT_POLYGON optional outputs? I think it's unlikely that models will usually require both outputs, and there's potentially a speed boost by not calculating unwanted convex hulls (not to mention the memory savings from avoiding creating the unwanted memory layers or cost of writing to a disk based format).

I'd leave both of them created by default (i.e. not taking advantage of the change in #4868)